app/services/import_service.rb

```diff
@@ -3,27 +3,35 @@
 class ImportService
   include Diaspora::Logging
 
-  def import_by_user(user, opts={})
-    import_by_files(user.export.current_path, user.exported_photos_file.current_path, user.username, opts)
+  def import_by_user(user, import_parameters)
+    profile_path = import_parameters["profile_path"]
+    photos_path = import_parameters["photos_path"]
+
+    import_by_files(user, profile_path, photos_path)
   end
 
-  def import_by_files(path_to_profile, path_to_photos, username, opts={})
-    if path_to_profile.present?
-      logger.info "Import for profile #{username} at path #{path_to_profile} requested"
-      import_user_profile(path_to_profile, username, opts.merge(photo_migration: path_to_photos.present?))
-    end
+  def import_by_files(user, path_to_profile, path_to_photos, opts={})
+    import_profile_if_present(opts, path_to_photos, path_to_profile, user.username)
+
+    import_photos_if_present(path_to_photos, user)
+    remove_import_files(path_to_profile, path_to_photos)
+  end
 
-    user = User.find_by(username: username)
-    raise ArgumentError, "Username #{username} should exist before uploading photos." if user.nil?
+  private
 
+  def import_photos_if_present(path_to_photos, user)
     if path_to_photos.present?
-      logger.info("Importing photos from import file for '#{username}' from #{path_to_photos}")
+      logger.info("Importing photos from import file for '#{user.username}' from #{path_to_photos}")
       import_user_photos(user, path_to_photos)
     end
-    remove_file_references(user)
   end
 
-  private
+  def import_profile_if_present(opts, path_to_photos, path_to_profile, username)
+    if path_to_profile.present?
```

Comment:
Style/GuardClause: Use a guard clause (`return unless path_to_profile.present?`) instead of wrapping the code inside a conditional expression.